### ISSUE TITLE
Sometimes the power-ups don't work

### ISSUE THREAD

**Issue by Valtosin** [Chaduke/paddles_balls_blocks_walls#4]
If you finish a level (lose or win) and pick up a power-up at that point, there's a chance that the power-ups won't work anymore.

**Comment by Chaduke** [comment]
Yes, there are lot of little of bugs to iron out with the power-ups I've been noticing.  This week my goal is to refactor a lot of the code I've just been banging out over the last week.   It's already quickly starting to look like a mess.  Part of it is that I've never used the Godot engine before buts its coming along just fine.  I stopped creating new stages at 10 for this specific reason though, because I want to make sure all the core functionality is working properly before adding more.  Will update when I make progress.  Thanks!

**Comment by Chaduke** [comment]
Hey big gameplay update today, check it out and let me know what you think.

**Comment by Valtosin** [comment]
Now everything is working as it should. Level 10 is too difficult and depends on randomness, it took about 10 minutes to get through it.

Also, if the framerate is low, the balls sometimes start flying through the platform.

**Comment by Chaduke** [comment]
This morning I was thinking about adding a "classic" ball physics mode where it behaves much more like the original game abd less like a real ball would, abd I think that would smooth out some of the issues like that with these levels.  The ball physics are largely being handled by the Godot engine with little tweaks by me here and there.  I like the realistic physics but I also like the other way as well, so I'll add this feature soon.  As far as frame rate goes, I have everything on the lowest compatibility settings and it's a web build so that likely is playing a part in that.  I don't notice it on my end, but I'm a about 8yr old machine but it was top of the line when I got it.  Do you mind sharing your system specs?  Processor speed, ram and gfx card?

**Comment by Valtosin** [comment]
@Chaduke, most likely it happens because there are a lot of balls on the map and the engine can't handle them. I think it's a webgl + chromium issue, considering I'm using a 12100f and an RTX 3070.

Maybe something like "[time dilation](https://en.wikipedia.org/wiki/Time_dilation)" attached to the physics engine could solve the problem, so that if there was an overload, the game would just slow down itself

**Comment by Chaduke** [comment]
I did speed up the physics updates in the beginning from 60 to 120 per second, because when I first started programming that kinda thing was happening a lot.  I don't really notice it much anymore. Maybe one ball at random sometimes.  With the new ball physics I'm bypassing the collision response and gravity that engine uses and I'm only using the detection, and it still happens from time to time.  I can get rid of it completely by doing a bounds check on every ball on every frame, but I haven't really gotten to the point where I'm optimizing and looking at performance yet.  I'll get there soon enough.   Right now I'm dealing with an issue where the web exporter is completely broken and I'm trying to figure out what happened, so for the last few days I'm basically not even getting to work on the game itself.
